src/v/net/connection.h

```diff
@@ -50,13 +52,27 @@ class connection : public boost::intrusive::list_base_hook<> {
     // NOLINTNEXTLINE
     const ss::socket_address addr;
 
+    /// Returns DN from client certificate
+    ///
+    /// The value can only be returned by the server socket and
+    /// only in case if the client authentication is enabled.
+    ss::future<std::optional<ss::session_dn>> get_distinguished_name() {
+        return _fd.get_distinguished_name();
+    }
+
+    const std::optional<security::tls::principal_mapper>&
+    get_principal_mapping() const {
+        return _tls_pm;
+    }
+
 private:
     boost::intrusive::list<connection>& _hook;
     ss::sstring _name;
     ss::connected_socket _fd;
     ss::input_stream<char> _in;
     net::batched_output_stream _out;
     server_probe& _probe;
+    std::optional<security::tls::principal_mapper> _tls_pm;
```

Comment:
GA: later when we go to GA we should probably explore pulling this out of `net::connection`. for example it'd be surprising if we ever used principal extraction here for RPC. so allowing it to be at the protocol level makes sense. but it works ok for now and is convenient.